### DISCUSSION TITLE
Simulating excavation or construction in time

### DISCUSSION THREAD

**Discussion by nlosacco** [idaholab/moose#23677] [Q&A Modules: General] [answered]
Dear all,

in the analysis of geotechnical problems, often I have to simulate an excavation or construction process. Due to non linearity, either processes need to be spread over a number of analysis increments.

A typical approach to the excavation problem involves:
1) removing elements and constraint the nodes at the new boundary (or apply nodal forces at the new boundary so that equilibrium is not altered)
2) incrementally reduce the reaction forces at the boundary to zero.

For construction:
1) activate new elements
2) incrementally change the material stiffness and gravity of the new elements from zero to the final value

I've found that elements can be removed or added to the analysis domain through e.g.:
CoupledVarThresholdElementSubdomainModifier
but is there any predefined way to do this "slowly" over a certain number of steps?

Best,

**Comment by GiudGiud** [comment, marked as answer]
Hello

While the process of changing an element from a subdomain to another is abrupt (as soon as the threshhold is passed), you can create BCs/material properties that depend continuously on the threshold or another variable to ramp the other quantities slower. 

I would use something like max(0, variable - threshold value) / (max_variable_value -  threshold value) as a multiplier

Guillaume

**Comment by PEI0214** [comment]
Hello, can you tell me the use 'CoupledVarThresholdElementSubdomainModifier' can simulate filling process? If possible, could you please provide more detailed instructions? It would be nice to have an example. Thank you for your help.

**Reply by GiudGiud** [reply]
Hello

you premesh the volume that will be filled, and you change its subdomain as the filling happens from an inactive subdomain (a subdomain with no equations) to an active one.

i think there are other posts with examples. There should be an app that has been developed to do this. @1runer do you know if that’s true?

**Reply by PEI0214** [reply]
Thank you for your reply. I want to zero the displacement calculated in the first step and then proceed to the next calculation, is there any good way to do it?

**Reply by GiudGiud** [reply]
Hello

You can only clear a variable using either a new simulation in which you do not restart that variable


Or you use a multiapp and set up a transfer that sends 0 values to that variable

Guillaume

**Reply by PEI0214** [reply]
Hello, Can I remove deformations only in part of the location? For example, the grid is divided into blocks 1 and 2, and the calculation continues after the deformation of block 2 is removed.
